OOPConsoleApp/Program.cs

```diff
@@ -0,0 +1,290 @@
+
+                Console.WriteLine($"Teacher - {teacher.PersonSurname} " +
+                    $"{teacher.PersonName}, " +
+                    $"{teacher.PersonAge} years old. ");
+                Console.WriteLine($"Temperament: {teacher.TeacherTemperament}. Subject: {teacher.TeacherSubject}.");
+            }
+            void Exercise3()
+            {
+                Console.WriteLine("Exercise 3.");
+
+                var personSecond = new Person("Datchin", "Valery", 43);
+
+                Console.WriteLine($"{personSecond.GetSurname()} {personSecond.GetName()}," +
+                $" {personSecond.GetAge()} y.o.");
+            }
+            void Exercise4()
+            {
+                Console.WriteLine("Exercise 4.");
+
+                var myPerson = new Person("Fleming", "Lise", 33);
+                myPerson.Print();
+
+                var myStudent = new Student("Koval", "Irina", 19);
+                myStudent.Print();
+
+                var myTeacher = new Teacher("Ivanov", "Ivan", 41);
+                myTeacher.Print();
+            }
+
+            void Exercise5()
+            {
+                Console.WriteLine("Exercise 5.");
+
+                var student1 = new Student("Pupkin", "Vasiliy", 20);
+                var student2 = new Student("Vasilyeva", "Iren", 18);
+
+                var teacher1 = new Teacher("Kirkorov", "Petya", 46);
+
+                var classroom = new Classroom(student1);
+
+                classroom.AddPerson(teacher1);
+                classroom.AddPerson(student2);
+
+                classroom.Print();
+
+                classroom.RemovePerson(student1);
+                classroom.RemovePerson(teacher1);
+
+                classroom.ClearPersons();
+            }
+        }
+    }
+}
+
+namespace OOPConsoleApp
+{
+    public class Classroom : IPrintable
+    {
+        private Dictionary<int, Person> _persons = new();
+        private Person _person;
+
+        public Classroom(Person person)
+        {
+            if (person is Student
+                || person is Teacher)
+            {
+                _person = person;
+                AddPerson(_person);
+            }
+        }
+
+        public void AddPerson(Person person)
+        {
+            var surname = person.PersonSurname;
+            var name = person.PersonName;
+            var age = person.PersonAge;
+
+            if (IsValid(surname, name, age))
+            {
+                person = new Person(surname, name, age);
+                _persons.Add(_persons.Count + 1, person);
+            }
+            else
+                throw new Exception("A person's data cannot be null!");
+        }
+
+        public bool IsValid(string surname, string name, int age)
+        {
+            if (!string.IsNullOrEmpty(surname)
+                && !string.IsNullOrEmpty(name))
+                return true;
+
+            return false;
+        }
+
+        public void RemovePerson(Person person)
```

Comment:
rework the deletion as we did the deletion from the repositories

Reply:
?+

Reply:
Now try to create another class where you would use this delete method.
For example, you have an add window (everything is ok there).
And there is a delete window, it will be a different class and a different file. To delete users, will you re-create objects?